lib/puppet/provider/package/cisco.rb

```diff
@@ -159,8 +208,13 @@ def in_guestshell?
   end
 
   def install
-    if in_guestshell? && target_host?
-      debug 'Guestshell + target=>host detected, using nxapi for install'
+    if (in_nexus? && in_guestshell? && target_host?) ||
+       (in_ios_xr? && target_host?)
+      if in_nexus?
+        debug 'Guestshell + target=>host detected, using nxapi for install'
+      elsif in_ios_xr?
+        debug 'target=>host detected, using sdr_instcmd for install'
+      end
       if @resource[:source]
         Cisco::Yum.install(@resource[:source])
       else
```

Comment:
Have you tested both conditional branches (with and without `@resource[:source]`) on IOS XR?